src/rgw/rgw_lc_s3.cc

```diff
@@ -32,7 +32,9 @@ bool LCExpiration_S3::xml_end(const char * el) {
   } else {
     date = lc_date->get_data();
     //We need return xml error according to S3
-    if (boost::none == ceph::from_iso_8601(date)) {
+    boost::optional<ceph::real_time> expiration_date = ceph::from_iso_8601(date);
+    timespec expiration_time = ceph::real_clock::to_timespec(*expiration_date);
```

Comment:
better off dereferencing `boost::optional<>` after checking it by comparing it with `boost::none`.

Reply:
Did you mean add `expiration_date = boost::none;`  after comparing?
Why better off dereferncing here?

Reply:
Yeah, I also want to know why? It is just a local variable.

Reply:
if `from_iso_8601()` returns `boost::none`, `ceph::real_clock::to_timespec(*expiration_date)` will trigger an assertion failure.

Reply:
yes, thanks. I'll revise it later.